### PR TITLE
docs: mark sdkVersion manifest property as deprecated

### PR DESCRIPTION
**Pull request by byCedric**
# Why

See #7574

I'm not 100% sure if we should add this to all versions, or if this is taken care of once `app.config.js`'s documentation is ready.

# How

With my magic keyboard

# Test Plan

Just docs.

### CODE REVIEW

File: docs/pages/versions/unversioned/workflow/configuration.md [hunk 1]
```diff
@@ -52,7 +52,9 @@ Valid values: `public`, `unlisted`
 
 ### `"sdkVersion"`
 
-**Required**. The Expo sdkVersion to run the project on. This should line up with the version specified in your package.json.
+__Deprecated__. The Expo sdkVersion to run the project on. This should line up with the version specified in your package.json.
+
+> This property is deprecated, it uses the `package.json`'s `expo` version to determine the SDK version. If you upgraded with `expo upgrade`, this property should have been removed.
```
Comment (SimenB) [inline docs/pages/versions/unversioned/workflow/configuration.md:57]: does it take the version from `package.json` or the version actually installed in `node_modules?

Comment (EvanBacon) [reply docs/pages/versions/unversioned/workflow/configuration.md:57]: It's currently only being read when the packages are installed. We should add support for reading from the `package.json` when packages aren't installed.

File: docs/pages/versions/unversioned/workflow/configuration.md [hunk 2]
```diff
@@ -52,7 +52,9 @@ Valid values: `public`, `unlisted`
 
 ### `"sdkVersion"`
 
-**Required**. The Expo sdkVersion to run the project on. This should line up with the version specified in your package.json.
+__Deprecated__. The Expo sdkVersion to run the project on. This should line up with the version specified in your package.json.
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/workflow/configuration.md:55]: This is now optional, we still require it for `UNVERSIONED`. Also I think @dsokal ran into some problems migrating turtle. Regardless we are still working on some of the smaller details regarding app.config.

Comment (byCedric) [reply docs/pages/versions/unversioned/workflow/configuration.md:55]: Sorry, I might have been to fast with this. I'll close this for now. I'll make a short checklist for me to double-check the documentation once that's done

Comment (dsokal) [reply docs/pages/versions/unversioned/workflow/configuration.md:55]: Actually, the only problem with Turtle was that turtle-cli couldn't find the `sdkVerison` field in `app.json`. It's later used for determining which shell app to pick. Other than that, it's not impacting Turtle. I fixed the issue by using `@expo/config` for finding the right version.

**Pull request closed without merging**